src/actions/clearMulticursors.ts

```diff
@@ -1,11 +1,47 @@
 import State from '../@types/State'
 import Thunk from '../@types/Thunk'
+import expandThoughts from '../selectors/expandThoughts'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
+import getThoughtById from '../selectors/getThoughtById'
+import head from '../util/head'
 
-/** Clears all multicursors. */
+/** Clears all multicursors and updates expanded state based on current cursor and previously selected thoughts. */
 const clearMulticursors = (state: State): State => {
+  // Get all previously selected paths
+  const selectedPaths = Object.values(state.multicursors)
+
+  // Create a new state without multicursors
+  const stateWithoutMulticursors = {
+    ...state,
+    multicursors: {},
+  }
+
+  // Expand based on cursor first
+  let newExpanded = expandThoughts(stateWithoutMulticursors, state.cursor)
```

Comment:
Can you explain why `newExpanded` is not enough on its own?

`expandThoughts(state, state.cursor)` should generally create the entire expansion state from the given state. Maybe I'm missing something though.

Reply:
The key is that this function is handling the cleanup of multicursors while preserving the expanded state of previously selected paths. 

Here's why both parts are needed.
`expandThoughts`(stateWithoutMulticursors, state.cursor) expands the path to the current cursor location.
However, the user might have previously expanded other paths while they had multiple selections (multicursors). 
If we only kept the cursor's expansion, those other expanded sections would collapse when clearing multicursors, which would be jarring for the user.

Reply:
Thanks. Can you provide a test or steps to reproduce? I'm open to the idea, but since it was not something originally proposed I will have to see what it's like with and without this additional change.

My concern is that this custom logic can cause `state.expanded` to be different even with the same state, depending on whether `clearMulticursors` was called or not. `state.expanded` should be completely deterministic so that the expansion state is a pure function of the cursor, multicursor, and surrounding thoughts.

Reply:
Thanks. However, I don't think it would be possible to expand `c` in this case, since multicursor thoughts are always collapsed.